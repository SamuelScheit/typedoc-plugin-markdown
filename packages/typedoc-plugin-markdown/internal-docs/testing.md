---
title: Testing guide
description: Guidance of running tests and creating fixtures.
---

# Testing

## Running tests

```bash
npm run test
```

If you require to update the snapshots run the following command:

```bash
npm run test:update
```

## Test Fixtures

A series of snapshot tests are used to ensure the output of the plugin is consistent.

- Example code for fixtures are located at `/test/fixtures/src`.
- Fixtures output scenarios are setup using a config file (`/test/fixtures/config.ts`) that provides a mechanism to easily provide a combination of options.
