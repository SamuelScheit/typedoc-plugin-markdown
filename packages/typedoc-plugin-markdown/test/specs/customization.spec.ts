import { expectFileToEqual } from '@devtools/testing';

describe(`Customization`, () => {
  test(`should insert content from hooks and apply custom theme`, () => {
    expectFileToEqual('customize', 'members', ['README.md']);
  });

  test(`should action pre-render-async jobs`, () => {
    expectFileToEqual('customize', 'members', ['post-render.txt']);
  });

  test(`should action post-render-async jobs`, () => {
    expectFileToEqual('customize', 'members', ['post-render.txt']);
  });
});
