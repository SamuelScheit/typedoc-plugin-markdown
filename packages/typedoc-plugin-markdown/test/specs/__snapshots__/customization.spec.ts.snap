// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customization should action post-render-async jobs: (Output File Strategy "members") (Option Group "1") 1`] = `"post render success"`;

exports[`Customization should action pre-render-async jobs: (Output File Strategy "members") (Option Group "1") 1`] = `"post render success"`;

exports[`Customization should insert content from hooks and apply custom theme: (Output File Strategy "members") (Option Group "1") 1`] = `
"> TOP BANNER using \`page.begin\` hook

<div style="display:flex; align-items:center;">
  <img alt="My logo" src="https://placehold.co/100x50" style="margin-right: .5em;" />
  <em>Welcome to typedoc-stubs with a customised header partial!!</em>
</div>

# typedoc-stubs

> CONTENT BANNER using \`content.begin\` hook

Module commments

<Tag type="warning">**Experimental** \`my-package\` is under active development.</Tag>

Some other comments

## Functions

### someFunction()

> **someFunction**(): \`void\`

#### Returns

\`void\`

**Generated using \`page.end\` hook**
"
`;
